src/utils/components/SelectTypeahead/SelectTypeahead.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { FC, ReactNode, Ref, useRef, useState } from 'react';
+
+import {
+  MenuToggleElement,
+  Select,
+  SelectList,
+  SelectOption,
+  SelectOptionProps,
+} from '@patternfly/react-core';
+import { useNetworkingTranslation } from '@utils/hooks/useNetworkingTranslation';
+import { isEmpty } from '@utils/utils';
+
+import Toggle from './Toggle';
+import { filterOptions } from './utils';
+
+type SelectTypeaheadProps = {
+  canCreate?: boolean;
+  id: string;
+  newOptionComponent?: (inputValue: string) => ReactNode;
+  options: SelectOptionProps[];
+  placeholder: string;
+  selected: string;
+  setSelected: (newSelection: null | string) => void;
+};
+
+const SelectTypeahead: FC<SelectTypeaheadProps> = ({
+  canCreate = true,
```

Comment:
I think it makes more sense to make the default to not create a new component, than when using the component it would be able to use canCreate when you need it, rather then canCreate={false} when you don't need it